gapic/schema/wrappers.py

```diff
@@ -1994,3 +2139,43 @@ def with_context(self, *,
             },
             meta=self.meta.with_context(collisions=collisions),
         )
+
+    def build_address_allowlist_for_selective_gapic(self, *,
+                                                    method_names: Iterable[str],
+                                                    address_allowlist: Set['metadata.Address'],
+                                                    resource_messages: Dict[str, 'MessageType'],
+                                                    services_in_proto: Dict[str, 'Service'],
+                                                    ) -> None:
+        """Builds a set of Addresses of wrapper objects to be included in selective GAPIC generation.
+
+        This method is used to create an allowlist of addresses to be used to filter out unneeded
+        services, methods, messages, and enums at a later step.
+        """
+
+        for method in self.methods.values():
+            if method.ident.proto in method_names:
+                # Include this service if there are any types/methods in selective gapic for this service.
+                address_allowlist.add(self.meta.address)
```

Comment:
Isn't this allow-listing this service it it has any methods at all, not just allowlisted ones?

Should the logic be something like this pseudocode?
```python
for method in self.methods.values():
  if method in allowlist:
    allowlist.add(self)
    allowlist.add(method)
    method.build_allowlist()

Reply:
`method_names` is the allowlist. I'll make that clearer by renaming it to `method_allowlist` or `method_names_allowlist`.